lib_src/src/multirate_hifi/asrc/src_mrhf_asrc.c

```diff
@@ -45,55 +45,90 @@
-#define        ASRC_FS_RATIO_88_176                    (unsigned int)((ASRC_FS_88_F / ASRC_FS_176_F) * ASRC_NOMINAL_FS_SCALE)
-#define        ASRC_FS_RATIO_88_192                    (unsigned int)((ASRC_FS_88_F / ASRC_FS_192_F) * ASRC_NOMINAL_FS_SCALE)
-
-#define        ASRC_FS_RATIO_96_44                        (unsigned int)((ASRC_FS_96_F / ASRC_FS_44_F) * ASRC_NOMINAL_FS_SCALE)
-#define        ASRC_FS_RATIO_96_48                        (unsigned int)((ASRC_FS_96_F / ASRC_FS_48_F) * ASRC_NOMINAL_FS_SCALE)
-#define        ASRC_FS_RATIO_96_88                        (unsigned int)((ASRC_FS_96_F / ASRC_FS_88_F) * ASRC_NOMINAL_FS_SCALE)
-#define        ASRC_FS_RATIO_96_96                        (unsigned int)((ASRC_FS_96_F / ASRC_FS_96_F) * ASRC_NOMINAL_FS_SCALE)
-#define        ASRC_FS_RATIO_96_176                    (unsigned int)((ASRC_FS_96_F / ASRC_FS_176_F) * ASRC_NOMINAL_FS_SCALE)
-#define        ASRC_FS_RATIO_96_192                    (unsigned int)((ASRC_FS_96_F / ASRC_FS_192_F) * ASRC_NOMINAL_FS_SCALE)
-
-#define        ASRC_FS_RATIO_176_44                    (unsigned int)((ASRC_FS_176_F / ASRC_FS_44_F) * ASRC_NOMINAL_FS_SCALE)
-#define        ASRC_FS_RATIO_176_48                    (unsigned int)((ASRC_FS_176_F / ASRC_FS_48_F) * ASRC_NOMINAL_FS_SCALE)
-#define        ASRC_FS_RATIO_176_88                    (unsigned int)((ASRC_FS_176_F / ASRC_FS_88_F) * ASRC_NOMINAL_FS_SCALE)
-#define        ASRC_FS_RATIO_176_96                    (unsigned int)((ASRC_FS_176_F / ASRC_FS_96_F) * ASRC_NOMINAL_FS_SCALE)
-#define        ASRC_FS_RATIO_176_176                    (unsigned int)((ASRC_FS_176_F / ASRC_FS_176_F) * ASRC_NOMINAL_FS_SCALE)
-#define        ASRC_FS_RATIO_176_192                    (unsigned int)((ASRC_FS_176_F / ASRC_FS_192_F) * ASRC_NOMINAL_FS_SCALE)
-
-#define        ASRC_FS_RATIO_192_44                    (unsigned int)((ASRC_FS_192_F / ASRC_FS_44_F) * ASRC_NOMINAL_FS_SCALE)
-#define        ASRC_FS_RATIO_192_48                    (unsigned int)((ASRC_FS_192_F / ASRC_FS_48_F) * ASRC_NOMINAL_FS_SCALE)
-#define        ASRC_FS_RATIO_192_88                    (unsigned int)((ASRC_FS_192_F / ASRC_FS_88_F) * ASRC_NOMINAL_FS_SCALE)
-#define        ASRC_FS_RATIO_192_96                    (unsigned int)((ASRC_FS_192_F / ASRC_FS_96_F) * ASRC_NOMINAL_FS_SCALE)
-#define        ASRC_FS_RATIO_192_176                    (unsigned int)((ASRC_FS_192_F / ASRC_FS_176_F) * ASRC_NOMINAL_FS_SCALE)
-#define        ASRC_FS_RATIO_192_192                    (unsigned int)((ASRC_FS_192_F / ASRC_FS_192_F) * ASRC_NOMINAL_FS_SCALE)
+#define		ASRC_FS_RATIO_44_44 						(uint32_t)((unsigned long long)(((double)ASRC_FS_44_F / ASRC_FS_44_F) * ((unsigned long long)1 << (ASRC_FS_RATIO_UNIT_BIT + 32))) >> 32)
+#define		ASRC_FS_RATIO_44_44_LO						(uint32_t)((unsigned long long)(((double)ASRC_FS_44_F / ASRC_FS_44_F) * ((unsigned long long)1 << (ASRC_FS_RATIO_UNIT_BIT + 32))))
```

Comment:
I don't quite get this - is one high precision and the other low? 

Scratch that, I have worked it out. High/low 32b of the 64b word..

Reply:
Yes, I wanted to make as few changes as possible so saved high and low separately so that just the high can still be used in places like calculating the max allowed deviation.